### PR TITLE
Added ⚡️ content to `Private Key Management` > `External Signing Devices`

### PR DESCRIPTION
**Pull request by Bosch-0**
:zap: [Live preview](https://deploy-preview-555--sad-borg-390916.netlify.app/guide/private-key-management/external-signers/)

Closes https://github.com/BitcoinDesign/Guide/issues/546

- Added LN content where relevant.
- Re-framed this page to focus on 'External Signers' as a concept not **just** hardware wallets. Changed page name to `External Signers` from `External Signing Devices` to reflect this. 
- Added image to 'How it works' section.
- Other minor formatting changes

### CODE REVIEW

File: guide/private-key-management/external-signer.md
Comment (GBKS) [inline guide/private-key-management/external-signer.md:51]:
```suggestion
   retina = "/assets/images/guide/private-key-management/external-signers/external-signer@2x.jpg"
```

Comment (GBKS) [reply guide/private-key-management/external-signer.md:51]: The three icons don't quite match. Different line thicknesses and levels of detail.

File: guide/private-key-management/external-signer.md
Comment (GBKS) [inline guide/private-key-management/external-signer.md:62]:
```suggestion
The external signer then signs the PSBT and sends it back to the primary wallet application. The users primary wallet application now broadcasts this now signed transaction to the Bitcoin network for processing. This process can also happen fully air-gapped using QR codes or memory cards instead of cables or wireless connections, to keep the device offline at all times.
```

File: guide/private-key-management/external-signer.md
Comment (GBKS) [inline guide/private-key-management/external-signer.md:68]:
```suggestion
An experimental type of external signers called Lightning signers that interact with the Lightning network also exist. Like regular external signers, Lightning signers isolate private keys from the users primary wallet application. However, these signers need to remain online and connected 24/7 to the users primary wallet application in order to update payment channel states. These, in effect, have some of the security benefits of regular external signers by isolating private keys from the internet and/or potential malicious device. Learn more [here](https://gitlab.com/lightning-signer/docs/-/blob/master/README.md). 
```

File: guide/private-key-management/external-signer.md
Comment (GBKS) [inline guide/private-key-management/external-signer.md:10]:
```suggestion
 - /guide/private-key-management/external-signing-device/
```

**Review by GBKS — CHANGES_REQUESTED** [review]
Great update, thank you for tackling this.

One future addition I could see is that of a smart auto-signer based on specific rules (auto-sign low amounts, based on allowances, etc).

For the list of products at the bottom, have there been any changes? Muun, for example, is a 2-of-2 with their service functioning as an auto-signer.

File: guide/private-key-management/external-signer.md
Comment (Bosch-0) [reply guide/private-key-management/external-signer.md:51]: Will fix now :)

**Comment by Bosch-0** [comment]
Removed that image in favour of adding a more detailed carousel (https://github.com/BitcoinDesign/Guide/issues/561) showing how this scheme works, thoughts @GBKS?

File: guide/private-key-management/external-signer.md
Comment (pavlenex) [inline guide/private-key-management/external-signer.md:41]:
```suggestion
External signers use, store, and manage private keys and other sensitive info, such as the user's recovery phrase, externally from the user's wallet application. The most common external signers used today are hardware wallets. Software-only external signing applications also exist, but are less secure than their hardware counterparts. 
```

File: guide/private-key-management/external-signer.md
Comment (pavlenex) [inline guide/private-key-management/external-signer.md:43]:
```suggestion
A wallet application supporting external signers removes the need for storing private keys locally by the wallet application. This decreases the theft risk as the keys can be isolated from the internet and/or a potentially malicious device. 
```

File: guide/private-key-management/external-signer.md
Comment (pavlenex) [inline guide/private-key-management/external-signer.md:51]: Is this always the case? I don't think it's how all external signers work? Hardware wallets for example don't use PSBT? They sign a transaction on the internal secure element and spit out the signature?

File: guide/private-key-management/external-signer.md
Comment (pavlenex) [inline guide/private-key-management/external-signer.md:59]:
```suggestion
Lightning signers are an experimental type of external signers interacting with the Lightning network. Like regular external signers, Lightning signers isolate private keys from the user's primary wallet application. However, these signers need to remain online and connected 24/7 to the user's primary wallet application to update payment channel states. These, in effect, have some of the security benefits of regular external signers by isolating private keys from the internet and/or potential malicious device. Learn more [here](https://gitlab.com/lightning-signer/docs/-/blob/master/README.md). 
```

File: guide/private-key-management/external-signer.md
Comment (pavlenex) [inline guide/private-key-management/external-signer.md:84]:
```suggestion
- When most users are new to Bitcoin and unlikely to implement good backup schemes
```

**Review by pavlenex — COMMENTED** [review]
Great page. Will need a bit of work once lighnting signers become the thing.

One thing that was unclear to me is the `PSBT` part. It isn't my impression that all external signers use PSBT, left comments on relevant places where I'd appreciate clarificaiton.

Besides that just a couple of minor nits. And of course the conflict in image compression that needs to be resolved 😺 

⚡

File: guide/private-key-management/external-signer.md
Comment (Bosch-0) [reply guide/private-key-management/external-signer.md:51]: Not all hardware wallets fully support PSBT yet that is true - though they really should! Coldcard, and Cobo are the only two that fully support it afaic. Pretty much every modern software external signer has full PSBT support though such as Nunchuk and Specter. The signing process doesn't always use a PSBT though - for example signing with a Trezor using Specter is done over USB using their own data structure: https://bitcoin.stackexchange.com/questions/100145/what-is-the-difference-between-psbt-support-in-coldcard-and-trezor

I'll reword this to make this more clear.

File: guide/private-key-management/external-signer.md
Comment (Bosch-0) [inline guide/private-key-management/external-signer.md:51]:
```suggestion
To sign a transaction, the primary wallet application submits a unsigned transaction or a partially signed bitcoin transaction ([PSBT](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki)) to the external signer.
```

File: guide/private-key-management/external-signer.md
Comment (Bosch-0) [inline guide/private-key-management/external-signer.md:53]:
```suggestion
The external signer then signs the transaction or PSBT and sends it back to the primary wallet application. The users primary wallet application now broadcasts this now signed transaction to the Bitcoin network for processing. This process can also happen fully air-gapped using QR codes or memory cards instead of cables or wireless connections, to keep the device offline at all times.
```

**Review by pavlenex — DISMISSED** [review]
Great work thanks for addressing the changes quickly 🚀

**Review by pavlenex — APPROVED** [review]

**Comment by pavlenex** [comment]
@GBKS This PR requires your re-review since it's blocked by `request a change` 😄

**Comment by GBKS** [comment]
All looks good now, thanks for the updates.

I noticed that the Overview page still refers to "External signing device" in the recommendation box, the diagrams, and the image and link under "Personal schemes". Can you address this here, or should we create a new PR for those tweaks?

**Comment by Bosch-0** [comment]
Can address is a new PR, will get on to that asap :)

**Review by pavlenex — APPROVED** [review]
LGTM, let's address the stuff @GBKS mentioned in a different PR.

**Comment by pavlenex** [comment]
@GBKS Needs your approval to get merged.

**Review by GBKS — APPROVED** [review]
Great update, especially with the reframing to get away from the device focus.

**Pull request merged**
